jerry-core/ecma/base/ecma-helpers-string.c

```diff
@@ -549,25 +549,18 @@ ecma_string_to_number (const ecma_string_t *str_p) /**< ecma-string */
     case ECMA_STRING_CONTAINER_MAGIC_STRING:
     case ECMA_STRING_CONTAINER_MAGIC_STRING_EX:
     {
-      const lit_utf8_size_t string_size = ecma_string_get_size (str_p);
+      ecma_number_t num;
+
+      ECMA_STRING_TO_UTF8_STRING (str_p, str_buffer_p);
 
-      if (string_size == 0)
+      if (str_buffer_p_size == 0)
```

Comment:
This `_size` variable is declared by the `ECMA_STRING_TO_UTF8_STRING` macro in a bit hidden way and used outside of it. (Ditto 5 lines below.)